webapi/Extensions/SemanticKernelExtensions.cs

```diff
@@ -152,12 +173,45 @@ private static Task RegisterSkillsAsync(IServiceProvider sp, IKernel kernel)
                 }
                 catch (SKException ex)
                 {
-                    var logger = kernel.LoggerFactory.CreateLogger(nameof(Kernel));
                     logger.LogError("Could not load skill from {Directory}: {Message}", subDir, ex.Message);
                 }
             }
         }
 
+        // Native skills
+        if (!string.IsNullOrWhiteSpace(options.NativeSkillsDirectory))
+        {
+            // Loop through all the files in the directory that have the .cs extension
+            var skillFiles = Directory.GetFiles(options.NativeSkillsDirectory, "*.cs");
```

Comment:
As you know, I'm not a big fan of having to ship the source files with the executable in order to know which classes are available..

Reply:
@glahaye can you expand a bit more on your reservations here? I'm curious how this applies to our webapi given that it is deployed as a zip

Reply:
Nobody is going to die because of this. But just imagine I told you that to deploy your executables on the cloud, you also needed to give me your source.  It's basically a hack.

Maybe (I haven't tested it out) a better way to go about that still uses reflection would be to look for functions that bear the "SKFunction" attribute.

Reply:
In order to not block this PR, I'm good with having a backlog item to revisit how we find the classes in the future.  Deal?  :)

Reply:
Reflection is slow either way, will leave this as is for simplicity in local testing. I've added another option to bypass this / serving source files where people can create their own setup hook and use that to import their custom skills at build time